pkg/http/listener.go

```diff
@@ -28,43 +28,32 @@ import (
 )
 
 var sslRequiredErrMsg = []byte("HTTP/1.0 403 Forbidden\r\n\r\nSSL required")
+var httpRequiredErrMsg = []byte("HTTP/1.0 403 Forbidden\r\n\r\nPlain-Text required")
+
+// The value chosen below is longest word chosen
+// from all the http verbs comprising of
+// "PRI", "OPTIONS", "GET", "HEAD", "POST",
+// "PUT", "DELETE", "TRACE", "CONNECT".
+const (
+	maxHTTPMethodLen = 7
```

Comment:
Because HTTP RFC doesn't dynamically change.  When new values are introduced we need to bump this value up. There is no reason to dynamically get this value.